src/common/blkdev.cc

```diff
@@ -219,6 +221,15 @@ int BlkDev::discard(int64_t offset, int64_t len) const
 
 bool BlkDev::is_rotational() const
 {
+  if (g_ceph_context != nullptr) {
```

Comment:
@brucen1030 hi Alice, instead of reading the setting from `g_ceph_context`, probably a better way is to do this at a higher level, for instance, in `KernelDevice::open()`?

Reply:
I understand your concern on g_ceph_context. 
In fact, BlueStore::is_rotational() is my first choice. I fixed it in BlkDev just to hide the details.
Do you mind me fixing it in BlueStore::is_rotational() directly?

Reply:
Putting it in BlueStore::is_rotational() is probably more explicit.

Reply:
Finally put it in KernelDevice::open() as suggested by Kefu, as BlueStore::is_rotational() can't cover everything... like OSD::update_crush_device_class().
In OSD initialization, OSD::update_crush_device_class() will call store->read_meta("crush_device_class", &device_class) and finally call BlockDevice::rotational which is initiated in KernelDevice::open() .